pulp_ansible/app/galaxy/views.py

```diff
@@ -52,8 +52,15 @@ def get_queryset(self):
         """
         Get the list of items for this view.
         """
-        distro = get_object_or_404(Distribution, base_path=self.kwargs['path'])
-        distro_content = distro.publication.repository_version.content
+        distro = get_object_or_404(AnsibleDistribution, base_path=self.kwargs['path'])
+
+        try:
+            distro_content = distro.repository_version.content
+        except AttributeError:
+            try:
+                distro_content = RepositoryVersion.latest(distro.repository).content
+            except AttributeError:
+                distro_content = []
```

Comment:
@bmbouter I want to ask you if this should maybe be something like:

```python
if distro.repository_version:
    distro_content = distro.repository_version.content
else:
    distro_content = RepositoryVersion.latest(distro.repository).content
```

I know this goes against Python's idiom of begging for forgiveness but lots of things could raise `AttributeError`. In fact I misspelled a method name and couldn't figure out the problem because it was being caught by this `except`. Is there some better way to do this?